Added contribution guide (#25)
nitindermohan/oak-new-docs: content/docs/contribution-guide/contributing-guide.md

---
title: "Contributing Overview"
summary: "How to contribute to Oakestra"
draft: false
weight: 401
toc: true
seo:
  title: "" # custom title (optional)
  description: "" # custom description (recommended)
  canonical: "" # custom canonical URL (optional)
  noindex: false # false (default) or true
---

> # Contributing Overview

Reviewer: ```suggestion
```
This title can be removed otw Hugo shows it twice
<img width="512" alt="image" src="https://github.com/user-attachments/assets/1aa15813-9072-49a4-9312-368b9cd68645">

**Welcome to Oakestra!** We're excited that you're interested in contributing to our project. This guide provides an overview of how to get started as a contributor. Whether you’re fixing bugs, improving documentation, or developing new features, your contribution is valued and important.

## Our Community Goals

> Oakestra is a welcoming, collaborative, and inclusive community. We want Oakestra to be a place where contributors feel empowered to share ideas, ask questions, and grow. All contributors are expected to follow our ~Code of Conduct~ to keep our community positive and open.

Reviewer: ```suggestion
Oakestra is a welcoming, collaborative, and inclusive community. We want Oakestra to be a place where contributors feel empowered to share ideas, ask questions, and grow. All contributors are expected to follow our _Code of Conduct_ to keep our community positive and open.
```

I guess you meant to underline the text?

The [Open Source Guides](https://opensource.guide/) website has a collection of resources for individuals, communities, and companies who want to learn how to run and contribute to an open source project.

Contributors and people new to open source alike will find the following guides especially useful.

{{< card-grid >}}
{{< link-card description="How to Contribute to Open Source" href="https://opensource.guide/how-to-contribute/" target="_blank" >}}
{{< link-card description="Building Welcoming Communities" href="https://opensource.guide/building-community/" target="_blank" >}}
{{< /card-grid >}}

## Code of Conduct

> Oakestra is committed to a respectful and inclusive environment. By contributing, you agree to abide by our [Code of Conduct](#link-to-code-of-conduct).

Reviewer: ```suggestion
Oakestra is committed to a respectful and inclusive environment. By contributing, you agree to abide by our [Code of Conduct](../code-of-conduct/).
```

## How You Can Contribute

> If you are eager to start contributing code right away, we have a list of [good first issues](https://github.com/oakestra/oakestra/labels/good%20first%20issue) that contain bugs which have a relatively limited scope. Alternatively, you can ceck for issues labeled as [“help wanted”](https://github.com/oakestra/oakestra/labels/help%20wanted) for beginner-friendly feature extensions. As you gain more experience and demonstrate a commitment to evolving Oakestra, you may be granted issue management permissions in the repository.

Reviewer: good first issues are usually the beginner-friendly ones, while help wanted all the rest. Also we need to specify how they can get the issue assigned. Here a suggestion:

```suggestion
If you are eager to start contributing code right away, we have a list of [good first issues](https://github.com/oakestra/oakestra/labels/good%20first%20issue) that contain bugs which have a relatively limited scope. Alternatively, you can ceck for issues labeled as [“help wanted”](https://github.com/oakestra/oakestra/labels/help%20wanted) for feature extensions that have not been handled yet. As you gain more experience and demonstrate a commitment to evolving Oakestra, you may be granted issue management permissions in the repository.

**How to contribute?** Just comment on an unassigned issue, stating you want to work on it. Once you get an ACK, you'll get assigned to a branch related to the issue. All your work must be merged to the assigned branch first via Pull Request. 
```
